libs/angular/src/components/settings/vault-timeout-input.component.ts

```diff
@@ -198,12 +199,24 @@ export class VaultTimeoutInputComponent
     this.vaultTimeoutPolicyHours = Math.floor(this.vaultTimeoutPolicy.data.minutes / 60);
     this.vaultTimeoutPolicyMinutes = this.vaultTimeoutPolicy.data.minutes % 60;
 
-    this.vaultTimeoutOptions = this.vaultTimeoutOptions.filter(
-      (t) =>
-        t.value <= this.vaultTimeoutPolicy.data.minutes &&
-        (t.value > 0 || t.value === VaultTimeoutInputComponent.CUSTOM_VALUE) &&
-        t.value != null,
-    );
-    this.validatorChange();
+    this.vaultTimeoutOptions = this.vaultTimeoutOptions.filter((vaultTimeoutOption) => {
+      // Always include the custom option
+      if (vaultTimeoutOption.value === VaultTimeoutInputComponent.CUSTOM_VALUE) {
+        return true;
+      }
+
+      if (typeof vaultTimeoutOption.value === "number") {
+        // Include numeric values that are less than or equal to the policy minutes
+        return vaultTimeoutOption.value <= this.vaultTimeoutPolicy.data.minutes;
+      } else {
+        // Exclude all string cases when there's a numeric policy defined
+        return false;
+      }
```

Comment:
```suggestion
      }
        
      // Exclude all string cases when there's a numeric policy defined
      return false;
```

Nit - you don't really need this `else` branch. The early returns above this statement make it irrelevant